mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb

```diff
@@ -156,6 +156,124 @@ func TestAccContainerNodePool_noName(t *testing.T) {
 	})
 }
 
+func TestAccContainerNodePool_withNoSpecifiedLoggingVariant(t *testing.T) {
+	t.Parallel()
+
+	cluster := fmt.Sprintf("tf-test-cluster-%s", randString(t, 10))
+	nodePool := fmt.Sprintf("tf-test-nodepool-%s", randString(t, 10))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerNodePoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccContainerNodePool_basic(cluster, nodePool),
+				Check: resource.TestCheckResourceAttr("google_container_node_pool.np",
+					"node_config.0.logging_variant",
+					"DEFAULT"),
+			},
+			resource.TestStep{
+				ResourceName:      "google_container_node_pool.np",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccContainerNodePool_withDefaultLoggingVariant(t *testing.T) {
+	t.Parallel()
+
+	cluster := fmt.Sprintf("tf-test-cluster-%s", randString(t, 10))
+	nodePool := fmt.Sprintf("tf-test-nodepool-%s", randString(t, 10))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerNodePoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccContainerNodePool_withLoggingVariant(cluster, nodePool, "DEFAULT"),
+				Check: resource.TestCheckResourceAttr("google_container_node_pool.with_logging_variant",
+					"node_config.0.logging_variant",
+					"DEFAULT"),
+			},
+			resource.TestStep{
+				ResourceName:      "google_container_node_pool.with_logging_variant",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccContainerNodePool_withMaxThroughputLoggingVariant(t *testing.T) {
+	t.Parallel()
+
+	cluster := fmt.Sprintf("tf-test-cluster-%s", randString(t, 10))
+	nodePool := fmt.Sprintf("tf-test-nodepool-%s", randString(t, 10))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckContainerNodePoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccContainerNodePool_withLoggingVariant(cluster, nodePool, "MAX_THROUGHPUT"),
+				Check: resource.TestCheckResourceAttr("google_container_node_pool.with_logging_variant",
+					"node_config.0.logging_variant",
+					"MAX_THROUGHPUT"),
+			},
+			resource.TestStep{
+				ResourceName:      "google_container_node_pool.with_logging_variant",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccContainerNodePool_withLoggingVariantUpdates(t *testing.T) {
```

Comment:
It could be worthwhile to just have this test for NodePool; it checks that the field is settable and also updatable and only creates one cluster for the test.

Reply:
Done.